core/trace.c

```diff
@@ -521,19 +505,15 @@ void tracepoint_scheduler_advancing_time_ends() {
  * @param worker The thread number of the worker thread or 0 for unthreaded execution.
  */
 void tracepoint_reaction_deadline_missed(reaction_t *reaction, int worker) {
-    tracepoint(reaction_deadline_missed, reaction->self, reaction->number, worker, NULL, NULL, 0);
+    tracepoint(reaction_deadline_missed, reaction->self, NULL, worker, worker, reaction->number, NULL, NULL, 0);
 }
 
-/**
- * Flush any buffered trace records to the trace file and
- * close the file.
- */
 void stop_trace() {
+    lf_mutex_lock(&_lf_trace_mutex);
```

Comment:
It doesn't seem right to possibly return (if `_lf_trace_stop`) without unlocking the mutex.